### PR TITLE
Add song of soaring screen.

### PR DESCRIPTION
**Pull request by ZoeyZolotova**

### CODE REVIEW

File: include/combo/oot/play.h [hunk 1]
```diff
@@ -26,7 +27,10 @@ typedef struct PACKED
 {
     char view[0x128];
     char unk_0128[0xdc88];
-    char textBuffer[0x556]; /* Probably smaller */
+    char textBuffer[0x550]; /* Probably smaller */
+    /* 0xE300 */ s32 msgLength;
+    /* 0xE304 */ u8 msgMode;
+    /* 0xE306 */ u8 unk_e305;
```
Comment (ZoeyZolotova) [inline include/combo/oot/play.h:33]: Comment should be E305 oops.

File: src/link_mm.in [hunk 2]
```diff
@@ -49,6 +49,7 @@ Shader_MoonTear             = 0x800ef65c;
 
 InitListPolyOpa = 0x8012c28c;
 InitListPolyXlu = 0x8012c2dc;
+Gfx_SetupDL_42Opa = 0x8012c8ac;
```
Comment (ZoeyZolotova) [inline src/link_mm.in:52]: This is a duplicate of `DrawDungeonUnk1`

File: src/oot/ovl/kaleido_scope.c [hunk 3]
```diff
@@ -150,3 +150,1340 @@ static void KaleidoScope_HandleDungeonChests(GameState_Play* play)
...
+    0xB1, // OWL_WARP_MOUNTAIN_VILLAGE
+    0xA9, // OWL_WARP_CLOCK_TOWN
+    0xB2, // OWL_WARP_MILK_ROAD
+    0xA8, // OWL_WARP_WOODFALL
+    0xB0, // OWL_WARP_SOUTHERN_SWAMP
+    0xAC, // OWL_WARP_IKANA_CANYON
+    0xAE, // OWL_WARP_STONE_TOWER
+};
+
+s16 sVtxPageMapWorldQuadsX[VTX_PAGE_MAP_WORLD_QUADS] = {
```
Comment (ZoeyZolotova) [inline src/oot/ovl/kaleido_scope.c:173]: Should probably be `static`. Probably in other places too.

**Comment by Onemario1234** [comment]
Just wanna say, this is insanely impressive!

File: src/oot/ovl/kaleido_scope.c [hunk 4]
```diff
@@ -150,3 +150,1340 @@ static void KaleidoScope_HandleDungeonChests(GameState_Play* play)
...
+            sUnpausedButtonStatus[3] = gSaveContext.buttonStatus[3];
+            sUnpausedButtonStatus[4] = gSaveContext.buttonStatus[4];
+
+            pauseCtx->map_x = 0;
+            pauseCtx->map_hilite = pauseCtx->map_cursor = pauseCtx->dungeonMapSlot = VREG(30) + 3;
+
+            WREG(16) = -175;
+            WREG(17) = 155;
+
+            u32 src = 0x7BD000;
```
Comment (ZoeyZolotova) [inline src/oot/ovl/kaleido_scope.c:620]: Would probably be cleaner to use kComboDmaData here too, even though it's a native file.

**Pull request merged**